config/jobs/dependency-watchdog/dependency-watchdog-kind-unit-tests.yaml

```diff
@@ -0,0 +1,65 @@
+presubmits:
+  gardener/dependency-watchdog:
+    - name: pull-dependency-watchdog-kind-unit-tests
```

Comment:
"kind unit tests" sounds like a contradiction in itself to me.
I expect unit tests not to have any external dependencies like a running k8s cluster (ref [testing guidelines](https://github.com/gardener/gardener/blob/master/docs/development/testing.md)).

Are these really "unit tests"? If not, please align the job name with our testing guidelines.
If they are "unit tests", then please reconsider the testing architecture in DWD.

Reply:
> "kind unit tests" sounds like a contradiction in itself to me.

I do not necessarily agree to this but i will make the change to follow the naming convention. We do only start a vanilla KIND cluster and do not need any gardener components.

Reply:
Changes are pushed. PTAL.

Reply:
DWD mainly requires a functional KAPI, etcd and KCM. For rest of the deployments like MCM and CA we start dummy deployments and change their status and replicas dynamically. So we start a KIND cluster to test the e2e functionality of prober. For weeder (which was previously called endpoint) we do not need any KIND tests.

Reply:
Do you really need KCM? If not, you might be interested in envtest, which is way easier to execute locally.

Reply:
Not sure about the details, but if it's just about updating some status that normally KCM would do, you can also fake its behaviour in your test (by manually updating it).

Reply:
Ok if there are many concerns then we will continue using concourse pipeline for KIND cluster tests. For release we will anyways use concourse pipeline. I can then close this PR. Let me know.

Reply:
I would request that you allow us to create this job as these are e2e tests but without gardener setup. We will rework these tests and perhaps re-use what g/g provides so that we test against gardener KIND cluster. Similar to what what is done in [extension-shoot-oidc-service](https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/master/hack/test-e2e-provider-local.sh)

Reply:
I think nobody wanted to deny/block this PR. There was just some confusion about the naming which triggered the suggestions above. Please go ahead. cc @oliver-goetz

Reply:
Thanks @rfranzke 
I have removed the hold.